src/infer_response.cc

```diff
@@ -154,33 +167,43 @@ InferResponse::LoadFromSharedMemory(
     bi::managed_external_buffer::handle_t* tensor_handle_shm =
         reinterpret_cast<bi::managed_external_buffer::handle_t*>(
             response_shm.data_.get() + sizeof(ResponseShm));
+    {
```

Comment:
Why does the `PbTensor::LoadFromSharedMemory` block for tensor need scoped GIL held, but the `PbString::LoadFromSharedMemory` block for parameters doesn't?

If you have more context on the scoped gil usage that was previously here, I think this comment could be improved - (needs to be held because XYZ is also contending for the GIL?):
```
      // Need to acquire the GIL to avoid hangs 
```

Reply:
The `PbTensor::LoadFromSharedMemory` may [access Python object](https://github.com/triton-inference-server/python_backend/blob/r24.12/src/pb_tensor.cc#L667), which requires holding the GIL, but the parameters are all C++ objects, so GIL is not required.

I'll update the comment to better describe the GIL requirement.

Reply:
Updated comment: https://github.com/triton-inference-server/python_backend/pull/394/commits/1acb95691182a9b3ff0be3eaac72ba8b4dc03273